orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -193,6 +193,9 @@ def __init__(self, model, model_desc, optim_config, loss_fn=None, options=None):
                         break
                 assert dtype is not None, f"ONNX model with unknown output type ({o_desc.name})"
 
+        from torch.utils.cpp_extension import ROCM_HOME
+        self.is_rocm_pytorch = (True if ((torch.version.hip is not None) and (ROCM_HOME is not None)) else False)
```

Comment:
(Not necessarily in this PR...) Could we add something similar to ORTModule, around where it decides to construct a CUDAExecutionProvider?

https://github.com/microsoft/onnxruntime/blob/1a1dd4843d97423ef1d322d9e2eb5ec25322de3e/orttraining/orttraining/python/training/ortmodule.py#L360

It would also be great to get at least a simple ORTModule test running as part of the ROCm CI pipeline.

Reply:
ORTModule has it already: https://github.com/microsoft/onnxruntime/blob/master/orttraining/orttraining/python/training/ortmodule.py#L362-L363

Reply:
Yeah, in fact, I borrowed the snippet from Weixing's PR against ORTModule for Rocm.

Reply:
Oh fantastic, thanks!